### DISCUSSION TITLE
Kann man über API einen Wechselrichter neu starten lassen?

### DISCUSSION THREAD

**Discussion by 2000ede** [tbnobody/OpenDTU#1789] [Q&A] [answered]
Hallo,
kann ich über die API auch einen Wechselrichter neu starten lassen?
Danke für die Antwort!

https://www.opendtu.solar/firmware/web_api/

**Comment by MetaChuh** [comment, marked as answer]
@2000ede 

> kann ich über die API auch einen Wechselrichter neu starten lassen?

ja kannst du.
beispiel mit curl befehl, den du in windows/linux/mac terminal zum gleich testen aufrufen kannst:

`curl --request POST --data "data={'serial':'your_inverter_serial','restart':true}" --user "admin:your_password" "http://your_opendtu_ip/api/power/config"`

bitte alle `your_*`werte mit deinen werten (hm seriennummer, passwort, ip deiner dtu) ersetzen.

hope this helps, and greetings,
metachuh

**Reply by 2000ede** [reply]
Thanks!

**Comment by TheFoxy** [comment]
curl --request POST --data "data={'serial':'your_inverter_serial','restart':true}" --user "admin:your_password"  
funtioniert aber nicht!
Warum ?

**Reply by ghost** [reply]
Keine Ahnung.
Hast du auch: `"http://your_opendtu_ip/api/power/config"` mit angehängt?!?

**Reply by ghost** [reply]
Gestern war es zu spät; gerade versucht: bei mir klappt es.

**Reply by maxclaudi** [reply]
curl von [MetaChuh](https://github.com/MetaChuh) funktioniert.

Funktioniert auch:
`curl -u "admin:Passwort" http://xxx.xxx.xxx.xxx/api/power/config -d 'data={"serial":"1xxxxxxxxxxx","restart":true}'`

edit:
admin beibehalten.
Passwort durch eigenes Passowort ersetzen.
xxx.xxx.xxx.xxx -> IP OpenDTU
1xxxxxxxxxxx -> Serial Inverter

**Comment by Chavell3** [comment]
Hi,
Ich muss dieses topic noch mal hoch holen.
Ich versuche gerade auch die DTU per API call neuzustarten, aber ohne erfolg, beide Kommandos bewirken nichts.

Anscheinend geht das Kommando durch aber es tut sich nichts... die DTU startet nicht durch...
![image](https://github.com/user-attachments/assets/38800e13-f0ae-429f-a316-656191b4285b)

Hat jemand noch eine Idee wie ich es testen kann, warum es nicht geht?
Danke.

**Reply by maxclaudi** [reply]
Nicht zu oft und schnell Befehle nacheinander senden.
OpenDTU und Hoymiles Wechselrichter benötigen beide Zeit.
 
versuch mal sequenziell (mit etwa 2 sek dazwischen):
`curl -u "admin:Passwort" http://xxx.xxx.xxx.xxx/api/power/status`

`curl -u "admin:Passwort" http://xxx.xxx.xxx.xxx/api/power/config -d 'data={"serial":"1xxxxxxxxxxx","restart":true}'`

evtl. nach 2 sek noch einmal:
`curl -u "admin:Passwort" http://xxx.xxx.xxx.xxx/api/power/status`

-------------------------------------
was auf jeden Fall funktionieren sollte, ist das aus- und einschalten.

Ausschalten:
`curl -u "admin:Passwort" http://xxx.xxx.xxx.xxx/api/power/config -d 'data={"serial":"1xxxxxxxxxxx","power":false}'`

Einschalten:
`curl -u "admin:Passwort" http://xxx.xxx.xxx.xxx/api/power/config -d 'data={"serial":"1xxxxxxxxxxx","power":true}'`

Bedenken, dass OpenDTU und der Wechselrichter Zeit benötigen (3~10sek.) um den Vorgang einzuleiten und abzuschließen.
Bei Hoymiles Wechselrichter funktionieren oben angegebene cURL.

Auch hier nach dem Ausschalten mal großzügig Zeit vergehen lassen bevor wieder eingeschaltet wird.
Nicht zu schnell.
Dem Wechselrichter auch Zeit zum starten und Betrieb lassen.
Manche Wechselrichter reagieren schneller, manche langsamer.

Hoffe es hilft weiter. 

---------------
edit PS:
Das betrifft den Wechselrichter Neustart!
Oder möchtest Du die DTU neu starten?

Das wäre was anderes:
`curl -u "admin:Passwort" http://xxx.xxx.xxx.xxx/api/maintenance/reboot -d 'data={"reboot":true}'`

**Comment by Gelegenheitsbastler** [comment]
Ist das Thema noch aktuell?
Ich hatte das (bekannte) Problem, dass die Verbindung zum Wlan verloren geht. Dann kommt natürlich auch ein Befehl nicht mehr an. So habe ich das inzwischen gelöst. Seitdem gibt es keine Verbindungsabbrüche mehr. 
https://github.com/tbnobody/OpenDTU/discussions/2624
